src/util/prolog/prologResult.ts

```diff
@@ -0,0 +1,120 @@
+import * as ld from "lodash";
+
+// First Group will be the variable value. Doesn't fetch the variable names
+const prologAnswerPattern: string = " = (\\w*|\\[.*\\])?(,| ;|.)";
+
+export class PrologResult {
+	private readonly rawResults: string[];
+	private readonly results: Map<string, Map<number, string|boolean>>;
+
+	constructor(rawResults: string[]) {
+		this.rawResults = rawResults;
+		this.results = new Map();
+	}
+
+	public getResults(): Map<string, Map<number, string|boolean>> {
```

Comment:
What is this function supposed to do? Feels like there's a simpler way to achieve that, but I don't really understand what this function is needed for.

Reply:
This function is supposed to return the cleaned up results or (if they haven't been created yet) it will create them from the rawResults. Using Arrays and implementing your other hints it is a lot simpler now.